stdlib/contrib/bonitoo-io/hex/hex.go

```diff
@@ -0,0 +1,185 @@
+	func(ctx context.Context, args values.Object) (values.Value, error) {
+		var str string
+		v, ok := args.Get(conversionArg)
+		if !ok {
+			return nil, errMissingArg
+		} else if v.IsNull() {
+			return values.Null, nil
+		}
+		switch v.Type().Nature() {
+		case semantic.String:
+			str = v.Str()
+		case semantic.Int:
+			str = strconv.FormatInt(v.Int(), 16)
+		case semantic.UInt:
+			str = strconv.FormatUint(v.UInt(), 16)
+		case semantic.Float:
+			str = strconv.FormatFloat(v.Float(), 'f', -1, 64)
+		case semantic.Bool:
+			str = strconv.FormatBool(v.Bool())
+		case semantic.Time:
+			str = v.Time().String()
+		case semantic.Duration:
+			str = v.Duration().String()
+		case semantic.Bytes:
+			str = goHex.EncodeToString(v.Bytes())
+		default:
+			return nil, errors.Newf(codes.Invalid, "cannot convert %v to string", v.Type())
+		}
+		return values.NewString(str), nil
+	},
+	false,
+)
+
+var intConv = values.NewFunction(
+	"int",
+	convIntType,
+	func(ctx context.Context, args values.Object) (values.Value, error) {
+		var i int64
+		v, ok := args.Get(conversionArg)
+		if !ok {
+			return nil, errMissingArg
+		} else if v.IsNull() {
+			return values.Null, nil
+		}
+		switch v.Type().Nature() {
+		case semantic.String:
+			base := 16
+			s := v.Str()
+			if strings.HasPrefix(s, "0x") {
+				s = s[2:]
+			} else if strings.HasPrefix(s, "+0x") || strings.HasPrefix(s, "-0x") {
+				base = 0
+			}
+			n, err := strconv.ParseInt(s, base, 64)
+			if err != nil {
+				return nil, errors.Newf(codes.Invalid, "cannot convert string %q to int due to invalid syntax", v.Str())
+			}
+			i = n
+		case semantic.Int:
+			i = v.Int()
+		case semantic.UInt:
+			i = int64(v.UInt())
+		case semantic.Float:
+			i = int64(v.Float())
+		case semantic.Bool:
+			if v.Bool() {
+				i = 1
+			} else {
+				i = 0
+			}
+		case semantic.Time:
+			i = int64(v.Time())
+		case semantic.Duration:
+			i = int64(v.Duration().Duration())
+		default:
+			return nil, errors.Newf(codes.Invalid, "cannot convert %v to int", v.Type())
+		}
+		return values.NewInt(i), nil
+	},
+	false,
+)
+
+var uintConv = values.NewFunction(
+	"uint",
+	convUintType,
+	func(ctx context.Context, args values.Object) (values.Value, error) {
+		var i uint64
+		v, ok := args.Get(conversionArg)
+		if !ok {
+			return nil, errMissingArg
+		} else if v.IsNull() {
+			return values.Null, nil
+		}
+		switch v.Type().Nature() {
+		case semantic.String:
+			s := strings.TrimPrefix(v.Str(), "0x")
+			n, err := strconv.ParseUint(s, 16, 64)
+			if err != nil {
+				return nil, errors.Newf(codes.Invalid, "cannot convert string %q to uint due to invalid syntax", v.Str())
+			}
+			i = n
+		case semantic.Int:
+			i = uint64(v.Int())
+		case semantic.UInt:
+			i = v.UInt()
+		case semantic.Float:
+			i = uint64(v.Float())
+		case semantic.Bool:
+			if v.Bool() {
+				i = 1
+			} else {
+				i = 0
+			}
+		case semantic.Time:
+			i = uint64(v.Time())
+		case semantic.Duration:
+			i = uint64(v.Duration().Duration())
```

Comment:
What is the use case for the other cases besides string?

If I read the code `hex.int(v: 5)` I am not sure what I would expect as an answer. I think it is confusing that the functions only convert from hex if the input is a string. 

If we removed all the cases except for the string case does that break the use case?

Reply:
It just works on other types ... as the built-in` int()` fn works.

Reply:
@nathanielc if you think that I should better restrict the type of `v` param to string, I will. I guess that (from a developer POV) I can't ensure that it is a string, especially when used with a table input (`hex.toInt`,`hex.toUInt`). It then provides a reasonable default behaviour rather than a strict failure.